validator_manager/src/move_validators.rs

```diff
@@ -0,0 +1,1259 @@
+        )
+        .arg(
+            Arg::with_name(GAS_LIMIT_FLAG)
+                .long(GAS_LIMIT_FLAG)
+                .value_name("UINT64")
+                .help(
+                    "All created validators will use this gas limit. It is recommended \
+                    to leave this as the default value by not specifying this flag.",
+                )
+                .required(false)
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name(FEE_RECIPIENT_FLAG)
+                .long(FEE_RECIPIENT_FLAG)
+                .value_name("ETH1_ADDRESS")
+                .help(
+                    "All created validators will use this value for the suggested \
+                    fee recipient. Omit this flag to use the default value from the VC.",
+                )
+                .required(false)
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name(BUILDER_PROPOSALS_FLAG)
+                .long(BUILDER_PROPOSALS_FLAG)
+                .help(
+                    "When provided, all created validators will attempt to create \
+                    blocks via builder rather than the local EL.",
+                )
+                .required(false)
+                .possible_values(&["true", "false"])
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name(STDIN_INPUTS_FLAG)
+                .takes_value(false)
+                .hidden(cfg!(windows))
+                .long(STDIN_INPUTS_FLAG)
+                .help("If present, read all user inputs from stdin instead of tty."),
+        )
+}
+
+#[derive(Clone, PartialEq, Debug, Serialize, Deserialize)]
+pub enum Validators {
+    All,
+    Count(usize),
+    Specific(Vec<PublicKeyBytes>),
+}
+
+#[derive(Clone, PartialEq, Debug, Serialize, Deserialize)]
+pub struct MoveConfig {
+    pub src_vc_url: SensitiveUrl,
+    pub src_vc_token_path: PathBuf,
+    pub dest_vc_url: SensitiveUrl,
+    pub dest_vc_token_path: PathBuf,
+    pub validators: Validators,
+    pub builder_proposals: Option<bool>,
+    pub fee_recipient: Option<Address>,
+    pub gas_limit: Option<u64>,
+    pub password_source: PasswordSource,
+}
+
+impl MoveConfig {
+    fn from_cli(matches: &ArgMatches) -> Result<Self, String> {
+        let count_flag = clap_utils::parse_optional(matches, COUNT_FLAG)?;
+        let validators_flag = matches.value_of(VALIDATORS_FLAG);
+        let validators = match (count_flag, validators_flag) {
+            (Some(count), None) => Validators::Count(count),
+            (None, Some(string)) => match string {
+                "all" => Validators::All,
+                pubkeys => pubkeys
+                    .split(',')
+                    .map(PublicKeyBytes::from_str)
+                    .collect::<Result<Vec<_>, _>>()
+                    .map(Validators::Specific)?,
+            },
+            (None, None) => Err("Must supply either --{:VALIDATORS_FLAG} or --{:COUNT_FLAG}.")?,
```

Comment:
This format string is a little wonky

```suggestion
            (None, None) => Err(format!("Must supply either --{VALIDATORS_FLAG} or --{COUNT_FLAG}."))?,
```